### ISSUE TITLE
no-promise-without-done-fail warns incorrectly in some cases

### ISSUE THREAD

**Issue by davidje13** [tlvince/eslint-plugin-jasmine#165]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
Linting reports `An "it" that uses an async method should handle failure (use "done.fail")` even when `done.fail` is included (from PR #158).

**If the current behavior is a bug, please provide the steps to reproduce.**

```
describe('Foo', () => {
  it('bar', (done) => {
    fetch('baz.txt') // any asynchronous method
      .then(done, (err) => done.fail(err));
  });
});
```

**What is the expected behavior?**
No error in this situation. It should be the same as:

```
describe('Foo', () => {
  it('bar', (done) => {
    fetch('baz.txt') // any asynchronous method
      .then(done, done.fail);
  });
});
```

(which does not report an error)

The current workaround when the error needs to be parsed in some way before reporting requires:

```
describe('Foo', () => {
  it('bar', (done) => {
    fetch('baz.txt') // any asynchronous method
      .catch((err) => Promise.reject(myErrorProcessing(err)))
      .then(done, done.fail);
  });
});
```

**Please mention your node.js, eslint-plugin-jasmine and operating system version.**
node: 10.8.0
eslint-plugin-jasmine: 2.10.1
os: osx 10.13.6

**Comment by GreenGremlin** [comment]
This is also an issue for me. I'd rather not have to rewrite tests in an unintuitive way to accommodate this behavior.
